projects/openapi-io/src/denormalizers/denormalizeProperty.ts

```diff
@@ -0,0 +1,125 @@
+import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';
+import { FlatOpenAPIV3 } from '@useoptic/openapi-utilities';
+import { JsonSchemaSourcemap } from '../parser/sourcemap';
+import { logPointer } from './pointer';
+
+// TODO this does not handle allOf: [{type:'object'...}, {allOf: [type:'object']}]
```

Comment:
@acunniffe This is a possible, yet unlikely case that I haven't handled here. It's a little more complicated because:
- we'll need to treat the resultant `allOf` object as part of the parent check to see if you can merge it again

It's doable, but will take more time to implement (example below)

```yml
schema:
  allOf:
    - allOf:
        - type: object
          properties:
            a: string
        - type: object
          properties:
            b: string
    - type: object
      properties:
        a:
          type: string
```